sp.yml

```diff
@@ -0,0 +1,39 @@
+---
+- hosts: "{{ target | default('localhost') }}"
+  become: no
+  connection: ssh
+  gather_facts: true
+
+  vars_files:
+    - "inventory/host_vars/all"
+    - "inventory/group_vars/all"
+    - "inventory/group_vars/{{ env }}/vars.yml"
+    - "inventory/group_vars/{{ env }}/vault.yml"
+    - "inventory/group_vars/{{ pop }}/vars.yml"
+    - "inventory/group_vars/{{ pop }}/vault.yml"
+    - "inventory/group_vars/{{ plat }}/vars.yml"
+    - "inventory/group_vars/{{ plat }}/vault.yml"
+
+  pre_tasks:
+    - name: get my public IP
+      ipify_facts:
+      #validate_certs: no
+
+  roles:
+    - { role: update-os-pckgs, become: true }
+    - { role: python, become: true }
```

Comment:
We don't need to install python each time. Right?

Reply:
no you don't. however Ansible detects that it is installed and proceed without reinstalling it. The benefit is that you have a generic and complete play that can be reused on others playbooks.
The other way around, you need to check the Python installation every time a role has this dependency (to avoid stopping the deployment).